### PR TITLE
Added compatibility for update v1.3.0

### PR DESCRIPTION
**Pull request by AntonyHanna**
Changes:

- Removed swap colors to reflect the changes `PlayerSpecificSettings` as of v1.3.0

- Removed swing rating for the same reason as above

- Included autoRestart to the `playerSettings` portion of the Status object

- Adjusted some variables and comments to reflect the new names in v1.3.0

- Adjusted the `protocol.md` file to reflect the above changes

p.s. forgive me for if there are any mistakes or just issues with this, this is my first time working with others code and using pull requests

### CODE REVIEW

File: BeatSaberHTTPStatus/BeatSaberHTTPStatusPlugin.csproj [hunk 1]
```diff
@@ -12,8 +12,8 @@
     <TargetFrameworkVersion>v4.6.1</TargetFrameworkVersion>
     <FileAlignment>512</FileAlignment>
     <TargetFrameworkProfile />
-    <Version>1.7.0</Version>
-    <BeatSaberVersion>1.1.0</BeatSaberVersion>
+    <Version>1.8.0</Version>
```
Comment (opl-) [inline BeatSaberHTTPStatus/BeatSaberHTTPStatusPlugin.csproj]: Do not change the plugin version. The plugin version gets bumped in the release commit. Until then the version stays the same.

File: BeatSaberHTTPStatus/GameStatus.cs [hunk 2]
```diff
@@ -96,12 +95,12 @@ public class GameStatus {
 		// Player settings
 		public bool staticLights = false;
 		public bool leftHanded = false;
-		public bool swapColors = false;
 		public float playerHeight = 1.7f;
 		public bool disableSFX = false;
 		public bool reduceDebris = false;
 		public bool noHUD = false;
 		public bool advancedHUD = false;
+        public bool autoRestart = false;
```
Comment (opl-) [inline BeatSaberHTTPStatus/GameStatus.cs]: Use tabs for indentation.

File: BeatSaberHTTPStatus/Plugin.cs [hunk 3]
```diff
@@ -243,11 +243,12 @@ public class Plugin : IPlugin {
 
 				gameStatus.staticLights = playerSettings.staticLights;
 				gameStatus.leftHanded = playerSettings.leftHanded;
-				gameStatus.swapColors = playerSettings.swapColors;
 				gameStatus.playerHeight = playerSettings.playerHeight;
 				gameStatus.disableSFX = playerSettings.disableSFX;
+                gameStatus.reduceDebris = playerSettings.reduceDebris;
```
Comment (opl-) [inline BeatSaberHTTPStatus/Plugin.cs]: Use tabs for indentation.

File: protocol.md [hunk 4]
```diff
@@ -93,12 +93,12 @@ StatusObject = {
 	"playerSettings": {
 		"staticLights": Boolean, // Static lights
 		"leftHanded": Boolean, // Left handed
-		"swapColors": Boolean, // Swap saber colors
 		"playerHeight": Number, // Player's height
 		"disableSFX": Boolean, // Disable sound effects
 		"reduceDebris": Boolean, // Reduce debris
 		"noHUD": Boolean, // No text and HUDs
 		"advancedHUD": Boolean, // Advanced HUD
+		"autoRestart" : Boolean, //Auto Restart on Fail
```
Comment (opl-) [inline protocol.md]: Missing space after `//`

File: BeatSaberHTTPStatus/Plugin.cs (hunk 3 above)
Comment (opl-) [reply BeatSaberHTTPStatus/Plugin.cs]: Also,  good catch.

Comment (AntonyHanna) [reply BeatSaberHTTPStatus/Plugin.cs]: I'm not sure how to go about doing this as visual studio isn't showing this to be an issue and Github Desktop isn't detecting that any changes have been made even when redoing the line indentation using tabs.

File: BeatSaberHTTPStatus/GameStatus.cs (hunk 2 above)
Comment (AntonyHanna) [reply BeatSaberHTTPStatus/GameStatus.cs]: Same issue as `Plugin.cs`

File: BeatSaberHTTPStatus/Plugin.cs (hunk 3 above)
Comment (AntonyHanna) [reply BeatSaberHTTPStatus/Plugin.cs]: Don't worry about my last comment, found the issue with my visual studio settings and have pushed a new commit with the changes requested.

**Comment by opl-** [comment]
Turns out you reverted the plugin version to 1.7.0 instead of 1.7.1. I noticed it only after merging and decided to mess with the history a bit to fix that. Don't worry, all is good now. I lost 1.7.1 myself for only a few minutes.

**Comment by AntonyHanna** [comment]
Sorry about that! don't know how I managed to mess that up.

**Comment by opl-** [comment]
Nah, no worries. I tried to rush releasing this and ended up creating troubles for myself.

While testing it turned out that `Plugin.OnNoteWasFullyCut` was throwing a `NullReferenceException` due to the `CutScoreBuffer._cutScoreWithMultiplier` field no longer existing. It's reflection so it doesn't come up while building, but it still crashes and burns when you try cutting any box.

Already fixed that in a128c13, but in the future we should probably both try running our code before publishing it. ^^,

**Pull request merged**